### PR TITLE
docs: refactor code-blocks that were looping over `records` as a `dict` instead of `pydantic.BaseModel`

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

As recently discussed with @frascuchon we should return the `FeedbackDataset.records` as `pydantic.BaseModel`s instead of Python dictionaries, since those are more user friendly and we should keep the validation layer for future releases.

**Type of change**

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Refactor (change restructuring the codebase without changing functionality)
- [X] Documentation update

**How Has This Been Tested**

- [X] Re-run all the Jupyter Notebooks and scripts to make sure those work, even fixed some that were not working before (neither with the previous approach)

**Checklist**

- [X] I have merged the original branch into my forked branch
- [X] I made corresponding changes to the documentation

**Missing**

In `docs/_source/guides/llms/conceptual_guides/rm.md` there's a code-block that mentions triplets, and that's assuming that some fields are created in the `FeedbackDataset` while in the definition on a code-block above those are not created, is that an extra piece of code, or should we create the fields `response 1`, `response 2`, and `response 3` mentioned there?

### CODE REVIEW

File: docs/_source/guides/llms/conceptual_guides/rm.md [hunk 1]
```diff
@@ -192,21 +192,25 @@ triplets = []
...
+        continue
 
     # Ensure the response has been submitted (not discarded)
-    if record['responses'][0]['status'] == 'submitted':
+    response = record.responses[0]
+    if response.status == 'submitted':
 
         # Get the preferred response index from the feedback
-        preferred_index = record['responses']['values']['response_ranking']
+        preferred_index = response.values['response_ranking'].value
```
Comment (davidberenstein1957) [inline docs/_source/guides/llms/conceptual_guides/rm.md]: don't we normally use double quotes?

File: docs/_source/guides/llms/practical_guides/fine_tune.md [hunk 2]
```diff
@@ -71,11 +71,11 @@ feedback_dataset = rg.FeedbackDataset.from_huggingface("argilla/databricks-dolly
 
 data = {"instruction": [], "context": [], "response": []}
 for entry in feedback_dataset:
-    if entry.get("responses"):
-        res = entry["responses"][0]["values"]
-        data["instruction"].append(res["new-instruction"]["value"])
-        data["context"].append(res["new-context"]["value"])
-        data["response"].append(res["new-response"]["value"])
+    if entry.responses is not None and len(entry.responses) > 0:
```
Comment (davidberenstein1957) [inline docs/_source/guides/llms/practical_guides/fine_tune.md:74]: can't we just do `if entry.responses:`?

File: docs/_source/guides/llms/practical_guides/fine_tune.md [hunk 3]
```diff
@@ -212,7 +212,7 @@ import argilla as rg
 from datasets import Dataset
 
 feedback = rg.FeedbackDataset.from_argilla("pre-training")
-content = {"content": [rec.get("fields").get("content") for rec in feedback]}
+content = {"content": [rec.fields["content"] for rec in feedback]}
```
Comment (davidberenstein1957) [inline docs/_source/guides/llms/practical_guides/fine_tune.md:215]: should we not replace all `rec.attr["key"]` with `.get` methods?

**Review by davidberenstein1957 — COMMENTED** [review]
Hi, some small remarks. Looks great and thanks for the update call.

File: docs/_source/guides/llms/conceptual_guides/rm.md (hunk 1 above)
Comment (alvarobartt) [reply docs/_source/guides/llms/conceptual_guides/rm.md]: Yes I usually do, just that didn't want to change too many things in the original code so I kept everything as-is besides the `pydantic.BaseModel` aka `rg.FeedbackRecord` usage instead

File: docs/_source/guides/llms/practical_guides/fine_tune.md (hunk 3 above)
Comment (alvarobartt) [reply docs/_source/guides/llms/practical_guides/fine_tune.md:215]: Indeed we can!

File: docs/_source/guides/llms/practical_guides/fine_tune.md (hunk 2 above)
Comment (alvarobartt) [reply docs/_source/guides/llms/practical_guides/fine_tune.md:74]: Sure!

File: docs/_source/guides/llms/conceptual_guides/rm.md (hunk 1 above)
Comment (frascuchon) [inline docs/_source/guides/llms/conceptual_guides/rm.md]:
```suggestion
        preferred_index = response.values["response_ranking"].value
```

File: docs/_source/guides/llms/conceptual_guides/rm.md [hunk 4]
```diff
@@ -192,21 +192,25 @@ triplets = []
...
         less_preferred_index = 1 if preferred_index == 2 else 2
 
         # Construct the triplet and append to the list
         triplets.append({
-            'prompt': record['fields']['prompt'],
-            'preferred_response': record['fields'][f'response {preferred_index}'],
-            'less_preferred_response': record['fields'][f'response {less_preferred_index}'],
+            'prompt': record.fields['prompt'],
+            'preferred_response': record.fields[f'response {preferred_index}'],
+            'less_preferred_response': record.fields[f'response {less_preferred_index}'],
```
Comment (frascuchon) [inline docs/_source/guides/llms/conceptual_guides/rm.md]:
```suggestion
            'prompt': record.fields["prompt"],
            'preferred_response': record.fields[f"response {preferred_index}"],
            'less_preferred_response': record.fields[f"response {less_preferred_index}"],
```

File: docs/_source/guides/llms/practical_guides/fine_tune.md (hunk 3 above)
Comment (frascuchon) [inline docs/_source/guides/llms/practical_guides/fine_tune.md:215]:
```suggestion
content = {"content": [rec.fields.get("content") for rec in feedback]}
```

File: docs/_source/guides/llms/practical_guides/fine_tune.md (hunk 2 above)
Comment (frascuchon) [inline docs/_source/guides/llms/practical_guides/fine_tune.md:74]:
```suggestion
    if entry.responses:
```

File: docs/_source/guides/llms/practical_guides/collect_responses.md [hunk 5]
```diff
@@ -60,11 +60,11 @@ fig.show()
 # plot 2: distance between responses in rating question
 list_values = []
 for record_ix,record in enumerate(feedback):
-    if record["responses"] != []:
-        submitted = [r for r in record["responses"] if r["status"] == "submitted"]
+    if record.responses != []:
```
Comment (frascuchon) [inline docs/_source/guides/llms/practical_guides/collect_responses.md]:
```suggestion
    if record.responses:
```

File: docs/_source/guides/llms/practical_guides/collect_responses.md [hunk 6]
```diff
@@ -40,12 +40,12 @@ from collections import Counter, OrderedDict
 import plotly.express as px
 
 count_submitted = Counter()
-for record in feedback:
-    if record["responses"] != []:
-        submitted = [r for r in record["responses"] if r["status"] == "submitted"]
+for record in feedback.records:
+    if record.responses != []:
```
Comment (frascuchon) [inline docs/_source/guides/llms/practical_guides/collect_responses.md]:
```suggestion
    if record.responses:
```

File: docs/_source/guides/llms/examples/curating-feedback-instructiondataset.ipynb [hunk 7]
```diff
@@ -636,20 +636,19 @@
    "source": [
     "new_records = []\n",
     "for record in feedback:\n",
-    "    if record['responses'] == []:\n",
+    "    if record.responses is None or len(record.responses) == 0:\n",
```
Comment (frascuchon) [inline docs/_source/guides/llms/examples/curating-feedback-instructiondataset.ipynb]:
```suggestion
    "    if record.responses:\n",
```

**Review by frascuchon — APPROVED** [review]

**Pull request merged**